libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
You are thinking about it backwards. How can I test the code if it's tightly coupled to a *specific* application? It doesn't make sense. FYI I've already tried your approach and no, it's nothing but a pain in the ass and just plain bad coding.